NEWAPI.md

```diff
@@ -0,0 +1,202 @@
+
+Please keep the list of `generate_api` calls in alphabetical order.
+
+Step 5: Run generateapis.sh
+---------------------------
+
+This will clone both the `googleapis` and `toolkit` repos as
+subdirectories, or pull them if they already exist.
+
+This may take a while, and will spew a lot of output. If it fails
+with a Java exception, that's probably due to a configuration issue
+somewhere. You can try to fix it yourself, but it's probably worth
+reaching out at that point.
+
+Step 6: Commit just the changes for your API
+--------------------------------------------
+
+You may see changes for other APIs, and right now (August 2017)
+you'll also see some new files to do with the Natural Language beta
+API. Ignore those. Create a commit containing:
+
+- Your `generateapis.sh` change
+- The new directory under `apis`
+
+Push this change up to your fork on github. Don't create a PR yet.
+
+Step 7: Pull your change on Windows
+-----------------------------------
+
+Clone your fork on Windows if you haven't already, and pull your
+change. (I find `git fetch --all` handy here, then just checkout
+your new branch.)
+
+Step 8: Modify the API catalog
+------------------------------
+
+Edit `apis/apis.json`. Again, this is in alphabetical order - please keep it that way.
+You'll typically want JSON like this:
+
+```json
+{
+  "id": "FIXME",
+  "productName": "FIXME",
+  "productUrl": "FIXME",
+  "version": "1.0.0-alpha00",
+  "type": "grpc",
+  "description": "FIXME",
+  "tags": [ "FIXME_1", "FIXME_2" ],
+}
+```
+
+Fix everything with "FIXME". There's no set number of tags, but these are used for the NuGet package,
+so consider what users will search for.
+
+The version number of "1.0.0-alpha00" is used to indicate that there
+hasn't been a release of this API yet. It will be incremented to
+`1.0.0-alpha01` as part of the first release.
+
+If your project uses the IAM or long-running operations APIs, you'll need to add dependencies for those, e.g.
+
+```json
+"dependencies": {
+  "Google.LongRunning": "project",
+  "Google.Cloud.Iam.V1": "project"
+}
+```
+
+Look at other APIs for example values.
+
+Step 9: Generate project and solution files
+-------------------------------------------
+
+Run `generateprojects.sh`. This should create:
+
+- A solution file
+- Project files for the production project and the snippets
+
+If you have Visual Studio 2017 installed, now would be a good time
+to load the solution and check it at least builds.
+
+(Note: currently if your API uses long-running operations, it *won't* compile until you've added a partial class like [this](https://github.com/GoogleCloudPlatform/google-cloud-dotnet/blob/58a38a605df48f8ee70d36e41c948bbd1b3628f3/apis/Google.Cloud.Speech.V1/Google.Cloud.Speech.V1/CloudSpeechGrpcPartial.cs).
+We will be generating that code soon.)
+
+Step 10: Commit changes
+-----------------------
+
+Commit the change to `apis.json` and the generated files.
+
+Optional: Step 11: Create documentation
```

Comment:
I'm all for auto generating the docs.  If it's not much work I say go for it, if it is a fair amount of work I wouldn't worry about it.  I don't see this happening that much and a copy paste of an existing doc would work.